Filename too long, failed download and crash when delete

### Steps to reproduce

Downloaded empty media and cannot be played, crash when delete

### Expected behavior

Media can be downloaded, played and deleted

### Actual behavior

Only download entry is shown, empty media and black thumbnail. Entry cannot be deleted and will crash.

### LibreTube version

0.17.1

### Android version

13

### Other details

`java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:558)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	... 1 more
Caused by: java.nio.file.FileSystemException: /storage/emulated/0/Android/data/com.github.libretube/files/thumbnail/爆笑祖國！盤點十大中國造假擺拍救災事件，假新聞“演員”畫面流出，颱風蘇拉，習慣光天化日下公開造假，當韭菜完全不存在，中國水災洪災救援被指造假，小粉紅不能接受的殘酷事實，戰狼沉默，杜蘇芮，台灣，陸配: File name too long
	at sun.nio.fs.UnixFileSystemProvider.implDelete(UnixFileSystemProvider.java:244)
	at sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:108)
	at j$.nio.file.spi.FileSystemProvider$VivifiedWrapper.deleteIfExists(Unknown Source:6)
	at j$.nio.file.Files.deleteIfExists(Unknown Source:4)
	at com.github.libretube.ui.adapters.DownloadsAdapter$$ExternalSyntheticLambda1.onClick(SourceFile:54)
	at androidx.preference.PreferenceFragmentCompat$1.handleMessage(SourceFile:94)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7918)
	... 3 more
    `

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

--- a/app/src/main/java/com/github/libretube/ui/dialogs/DownloadDialog.kt
+++ b/app/src/main/java/com/github/libretube/ui/dialogs/DownloadDialog.kt
@@ -162,8 +162,14 @@ class DownloadDialog : DialogFragment() {
         restorePreviousSelections(binding, videoStreams, audioStreams, subtitles)
 
         onDownloadConfirm = onDownloadConfirm@{
-            if (binding.fileName.text.toString().isEmpty()) {
+            val fileName = binding.fileName.text.toString()
+            if (fileName.isBlank()) {
                 Toast.makeText(context, R.string.invalid_filename, Toast.LENGTH_SHORT).show()
+                return@onDownloadConfirm
+            }
+
+            if (fileName.length > MAX_FILE_NAME_LENGTH - 20) { // reserve 20 chars for quality and extension
+                Toast.makeText(context, R.string.filename_too_long, Toast.LENGTH_SHORT).show()
                 return@onDownloadConfirm
             }
 
@@ -269,6 +275,11 @@ class DownloadDialog : DialogFragment() {
     }
 
     companion object {
+        /**
+         * Max file name length at Android systems
+         */
+        private const val MAX_FILE_NAME_LENGTH = 255
+
         private const val VIDEO_DOWNLOAD_QUALITY = "video_download_quality"
         private const val VIDEO_DOWNLOAD_FORMAT = "video_download_format"
         private const val AUDIO_DOWNLOAD_QUALITY = "audio_download_quality"

--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -472,6 +472,7 @@
     <string name="segment_type">Segment type</string>
     <string name="sb_invalid_segment">Invalid segment start or end</string>
     <string name="contribute_to_sponsorblock">Contribute to SponsorBlock</string>
+    <string name="filename_too_long">Filename too long!</string>
 
     <!-- Notification channel strings -->
     <string name="download_channel_name">Download Service</string>

